hexadecimal/solution.py

```diff
@@ -1,10 +1,17 @@
-from functools import reduce
+_hex_code = dict(zip('0123456789', range(10)))
+_hex_code.update(zip('abcdef', range(10, 16)))
+_hex_code.update(zip('ABCDEFG', range(10, 16)))
```

Comment:
Another option would be to convert the input string to lowercase, then the uppercase letters would not be needed here.

Reply:
Right, but lowercasing has to happen for every function call, while the dict is generated only once at import time. Of course, performance is not an important criterion for this toy example, but why not opt in if this is all it takes.

Reply:
That's a fair point.